core/src/test/java/com/datastax/oss/driver/internal/core/type/codec/VectorCodecTest.java

```diff
@@ -49,21 +51,21 @@ public void should_encode() {
   /** Too few eleements will cause an exception, extra elements will be silently ignored */
   @Test
   public void should_throw_on_encode_with_too_few_elements() {
-    assertThatThrownBy(() -> encode(VECTOR.subList(0, 1)))
+    assertThatThrownBy(() -> encode(VECTOR.subVector(0, 1)))
         .isInstanceOf(IllegalArgumentException.class);
   }
 
   @Test
   public void should_throw_on_encode_with_empty_list() {
-    assertThatThrownBy(() -> encode(Lists.newArrayList()))
+    assertThatThrownBy(() -> encode(CqlVector.newInstance()))
         .isInstanceOf(IllegalArgumentException.class);
   }
 
   @Test
   public void should_encode_with_too_many_elements() {
-    List<Float> doubleVector = Lists.newArrayList(VECTOR);
-    doubleVector.addAll(VECTOR);
-    assertThat(encode(doubleVector)).isEqualTo(VECTOR_HEX_STRING);
+    Float[] doubledVectorContents = Arrays.copyOf(VECTOR_ARGS, VECTOR_ARGS.length * 2);
+    System.arraycopy(VECTOR_ARGS, 0, doubledVectorContents, VECTOR_ARGS.length, VECTOR_ARGS.length);
+    assertThat(encode(CqlVector.newInstance(doubledVectorContents))).isEqualTo(VECTOR_HEX_STRING);
```

Comment:
Why does encode not fail when encoding a 4-length vector with a 2-length codec? Is there a use-case for this?

Reply:
Short version is that the codec loops through the expected number of dimension, pulling off a ByteBuffer for the next element and attempting to encode that.  If there are additional bytes after that they're just silently ignored.

There is no real use case for this, and it shouldn't happen in regular operation (barring some weird server error).  I included it in the test here in an attempt to document the differing behaviour between getting too little data vs. too much data.

Reply:
Should we have the codec enforce exact length on the input vector? I spent some time puzzling why encoded bytes length-4 vector was the same as for the length-2 version. I can see this leading to unexpected behavior if lengths get mixed up in customer code

Reply:
There's not a clean answer here.  This isn't the kind of thing that customers will manipulate manually.  Normally we'll get a type from the server saying something like "this field is a vector of subtype float with 10 dimensions" and we'll create a codec for that in response.  If the server then sends garbage data for fields with that type we have a problem but the user isn't actually involved in that interaction.

I can see this both ways.  It's not unreasonable to say "you told me you were going to send 10 dimensions and you sent 20, that's an error".  It's also not unreasonable to say "you told me you were sending 10 dimensions and you sent 20... well, at least I can do my job (and ignore the rest)".

It's worth noting that ListCodec has a similar problem.  In the case of a list the number of elements is passed along in an int before the actual elements appear, but (hypothetically) that number could be off and the server could send more data.  ListCodec doesn't care; it tries to decode the number of elements it believes are in the list and returns that.  There's no validation that I can see saying "did you read everything?".  If any such validation _does_ exist it would have to be higher up in the stack... which means it would apply to CqlVector as well.